File: datastructures/graph/DFS.cpp
#include "DFS.hpp"

using namespace std;

DFS::DFS(Graph *g, int s) {
    marked = vector<bool> (g->getV(), false);
    dfs(g, s);
}

void DFS::dfs(Graph *g, int v) {
    marked[v] = true;
    count += 1;
    for(int w : g->getAdj()->at(v)) {
        if (!marked[w]) {
            dfs(g, w);
        }
    }
}

int DFS::getCount() {
    return count;
}

vector<bool>* DFS::getMarked() {
    return &marked;
}

File: datastructures/graph/dfs.js
"use strict";

class DFS {
    constructor(g, s) {
        this.marked = [];
        this.count = 0;
        for (let i = 0; i < g.V; i++ ) {
            this.marked.push(false);
        }
        this.dfs(g, s);
    }

    dfs(g, s) {
        this.marked[s] = true;
        this.count += 1;
        for (const v of g.adj[s]) {
            if (!this.marked[v]) {
                this.dfs(g, v);
            }
        }
    }
}

module.exports = DFS;
